<?xml version="1.0"?>

<robot name="inmoov" xmlns:xacro="http://ros.org/wiki/xacro">

<xacro:include filename="$(find inmoov_description)/urdf/materials.urdf.xacro" />

<xacro:macro name="head" params="covers">

  <link name="head_base">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/head_base.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <xacro:frame_color/>
    </visual>
  </link>

  <link name="head">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/head.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <xacro:frame_color/>
    </visual>
  </link>

  <link name="jaw">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/jaw.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 3.1415" xyz="0 0 0"/>
      <xacro:cover_color/>
    </visual>
  </link>

  <link name="skull">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/skull.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.001 0 0"/>
      <xacro:cover_color/>
    </visual>
  </link>

  <link name="left_ear">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/earleftv1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <xacro:frame_color/>
    </visual>
  </link>

  <link name="right_ear">
    <visual>
      <geometry>
        <mesh filename="package://inmoov_description/meshes/earrightv1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <xacro:frame_color/>
    </visual>
  </link>


  <joint name="head_leftright" type="revolute">
   <axis xyz="0 0 1"/>
   <limit effort="1000.0" lower="-1.57" upper="1.57" velocity="0.5"/>
   <origin rpy="0 0 0" xyz="0 0 0.0245"/>
   <parent link="head_base"/>
   <child link="head"/>
  </joint>

  <joint name="jaw" type="revolute">
   <axis xyz="1 0 0"/>
   <limit effort="1000.0" lower="0" upper="0.15" velocity="0.5"/>
   <origin rpy="0.085 0 3.1415" xyz="-0.001 0.0375 0.0437"/>
   <parent link="head"/>
   <child link="jaw"/>
  </joint>

  <joint name="skull" type="fixed">
    <origin rpy="0 0 0" xyz="-0.001 0.044 -0.0245"/>
    <parent link="head"/>
    <child link="skull"/>
  </joint>

  <joint name="left_ear" type="fixed">
    <origin rpy="0 0 0" xyz="-0.056 0.035 0.0437"/>
    <parent link="head"/>
    <child link="left_ear"/>
  </joint>

  <joint name="right_ear" type="fixed">
    <origin rpy="0 0 0" xyz="0.056 0.035 0.0437"/>
    <parent link="head"/>
    <child link="right_ear"/>
  </joint>



</xacro:macro>

</robot>
